### PR TITLE
Change ControlStateManager to be a singleton

### PR DESCRIPTION
**Pull request by toly-kournik**

### CODE REVIEW

File: bftengine/src/bftengine/ReplicaBase.hpp [hunk 1]
```diff
@@ -126,8 +121,7 @@ class ReplicaBase {
 
   ///////////////////////////////////////////////////
   // ControlStateManger
-  std::shared_ptr<ControlStateManager> controlStateManager_;
-  bool stopAtNextCheckpoint_ = false;
+  bool stopAtNextCheckpoint_ = false;  // TODO [TK] remove, use ControlStateManager's state
```
Comment (yontyon) [inline bftengine/src/bftengine/ReplicaBase.hpp:124]: I think we already have a cache variable in ControlStateMnager, so we can remove this variable and its uses

File: kvbc/include/KVBCInterfaces.h [hunk 2]
```diff
@@ -121,7 +121,6 @@ class ICommandsHandler : public bftEngine::IRequestsHandler {
   void execute(ExecutionRequestsQueue& requestList,
                const std::string& batchCid,
                concordUtils::SpanWrapper& parent_span) override = 0;
-  virtual void setControlStateManager(std::shared_ptr<bftEngine::ControlStateManager> controlStateManager) = 0;
```
Comment (yontyon) [inline kvbc/include/KVBCInterfaces.h:124]: Notice that this breaks API in concord also

**Review by yontyon — APPROVED** [review]
LGTM
Only 2 minor comments

File: kvbc/include/KVBCInterfaces.h (hunk 2 above)
Comment (toly-kournik) [reply kvbc/include/KVBCInterfaces.h:124]: Yes, we'll do it during bft version bump

File: bftengine/src/bftengine/ReplicaBase.hpp (hunk 1 above)
Comment (toly-kournik) [reply bftengine/src/bftengine/ReplicaBase.hpp]: Removed, just wanted your feedback

**Pull request merged**